test/05-device-state.spec.ts

```diff
@@ -355,4 +356,26 @@ describe('deviceState', () => {
 	it.skip('applies the target state for device config');
 
 	it.skip('applies the target state for applications');
+
+	it('prevents reboot or shutdown when HUP rollback breadcrumbs are present', async () => {
+		const testErrMsg = 'Waiting for Host OS updates to finish';
+		stub(updateLock, 'ensureNoHUPBreadcrumbsOnHost').throws(
+			new UpdatesLockedError(testErrMsg),
+		);
+
+		await deviceState
+			.reboot()
+			.then(() => expect.fail())
+			.catch((err: any) => {
+				expect(err.message).to.equal(testErrMsg);
+			});
+		await deviceState
+			.shutdown()
+			.then(() => expect.fail())
+			.catch((err: any) => {
+				expect(err.message).to.equal(testErrMsg);
+			});
```

Comment:
I think this will fail if the promise does not throw but also passes when it does throw and matches our error (also matches on the error type which is nice since some logic does err instanceof)

`
await expect(deviceState.reboot()).to.eventually.throw(new UpdatesLockedError(testErrMsg));
await expect(deviceState.shutdown()).to.eventually.throw(new UpdatesLockedError(testErrMsg));
`

Reply:
feel free to see if this cleans up any other test case code such as the update-lock spec

Reply:
`await expect(deviceState.shutdown()).to.eventually.throw(new UpdatesLockedError(testErrMsg));` doesn't work but removing the async/await pattern from the test case does:
```
expect(deviceState.reboot()).to.eventually.throw(new UpdatesLockedError(testErrMsg));
expect(deviceState.shutdown()).to.eventually.throw(new UpdatesLockedError(testErrMsg));
```

Verified that the stub restore call after these 2 expects is called as well. Chai-as-promised handles the underlying promise so no need for async/await I guess.

Reply:
That's weird, because some other test cases use async/await no problem for a setup similar to this..

Reply:
```
await expect(deviceState.reboot()).to.be.rejectedWith(testErrMsg);
await expect(deviceState.shutdown()).to.be.rejectedWith(testErrMsg);
```
works. Gotta remember to use `rejectedWith` instead of `throw` (note that `rejectedWith(new UpdatesLockedError(testErrMsg)` will not pass). I'm almost certain I've dealt with this exact scenario before with chai-as-promised while writing tests, which means I almost prefer the non chai-as-promised method I had originally:
```
await promise()
.then(() => expect.fail())
.catch(err => {
  expect(err).to.be.instanceOf(UpdatesLockedError);
  expect(err.message).to.equal(testErrMsg);
})
```

Reply:
Already committed to using chai-as-promised though because I spent time on it, so I'll change it 👍🏼